src/compute-pr-actions.ts

```diff
@@ -181,15 +186,21 @@ function extendPrInfo(info: PrInfo): ExtendedPrInfo {
     const hasChangereqs = changereqReviews.length > 0;
     const approvalFlags = getApprovalFlags();
     const approverKind = getApproverKind();
-    const approved = getApproval();
+    const firstApprovalDate = hasChangereqs ? undefined : earliestDate(...approvedReviews.map(r =>
+        r.isMaintainer
+        || approverKind === "owners" && allOwners.includes(r.reviewer)
+        || approverKind === "others"
+            ? r.date : undefined));

```

Comment:
My rule of thumb here is that an expression that is short enough to keep it a one-liner is fine, and more complicate things (like `approved`) are pushed to a `getX` function at the end.

Reply:
Done :heavy_check_mark: